src/wifi/model/periodic-dmg-wifi-scheduler.cc

```diff
@@ -0,0 +1,490 @@
+    }
+  else if (info.GetAllocationFormat () == ASYNCHRONOUS)
+    {
+      // for asynchronous allocations, the Maximum Allocation field is reserved (IEEE 802.11ad 8.4.2.136) 
+      newDuration = dmgTspec.GetMinimumAllocation ();
+    }
+  else
+    {
+      NS_FATAL_ERROR ("Allocation Format not supported");
+    }
+
+  AllocationFieldListI allocation;
+  // Retrieve the allocation for which a modification has been requested
+  for (allocation = m_addtsAllocationList.begin (); allocation != m_addtsAllocationList.end ();)
+    {
+      if ((allocation->GetAllocationID () == info.GetAllocationID ())
+          && (allocation->GetSourceAid () == sourceAid) 
+          && (allocation->GetDestinationAid () == info.GetDestinationAid ()))
+        {
+          break;
+        }
+      else
+        {
+          ++allocation;
+        }
+    }
+
+  NS_ABORT_MSG_IF (allocation == m_addtsAllocationList.end (), "Required allocation does not exist.");
+
+  uint32_t currentDuration = allocation->GetAllocationBlockDuration ();
+  NS_LOG_DEBUG ("current duration=" << currentDuration << ", new duration=" << newDuration);
+  uint32_t timeDifference;
+  if (newDuration > currentDuration)
+    {
+      NS_LOG_DEBUG ("The increase in slot duration is not supported by this version of PeriodicDmgWifiScheduler.");
+      // The request cannot be accepted; maintaining old allocation duration
+      // No need to update allocation start time and remaining DTI time 
+      status.SetFailure ();
+    }
+  else
+    {
+      NS_LOG_DEBUG ("Reduction of the duration is always allowed. Proceed to update the available slots...");
+      timeDifference = currentDuration - newDuration;
+      allocation->SetAllocationBlockDuration (newDuration);
+      status.SetSuccess ();
+
+      uint32_t startAlloc = allocation->GetAllocationStart ();
+      uint32_t endAlloc = allocation->GetAllocationStart () + allocation->GetAllocationBlockDuration () + m_guardTime;
+      uint16_t allocPeriod = allocation->GetAllocationBlockPeriod ();
+
+      // If the number of blocks is > 0 we have to update the available slots in the DTI accordingly.
+      // TODO: update also the number of blocks if the new duration allows to 
+      // add further blocks
+      uint8_t blocks = allocation->GetNumberOfBlocks ();
+      for (uint8_t i = 0; i < blocks; ++i)
+        {
+          NS_LOG_DEBUG ("Modify SP Block: starts at " << startAlloc << " and lasts till " << endAlloc);
+          UpdateAvailableSlots (startAlloc, endAlloc, timeDifference);
+          startAlloc += allocPeriod;
+          endAlloc += allocPeriod;
+        }
+    }
+  return status;
+}
+
+void
+PeriodicDmgWifiScheduler::AddBroadcastCbapAllocations (void)
+{
+  NS_LOG_FUNCTION (this);
+
+  // Addts allocation list is copied to the allocation list 
+
+  m_allocationList = m_addtsAllocationList;
+  AllocationFieldList broadcastCbapList;
+
+  // fill all the remaining available slots with broadcast CBAPs
```

Comment:
At this point, are `m_availableSlots` and `m_addtsAllocationList`already ordered? 
Is there a case when they can be unordered? If no, I think we can avoid the sort at the end and insert elements in `m_allocationsList` using an incrementing iterator.

Reply:
`m_addtsAllocationList` can be unordered, as discussed offline, and this is reflected to `m_allocationList`. By construction, instead, `m_availableSlots` is always ordered, which is the reason why we can do the for-loop in this method. 
The sort at the end of the function is done just for my own mental sanity: when debugging, it is easier to read an ordered list in order to understand if everything is working properly. Having said that, yes, if you want we can delete it.

Reply:
Sure, you told me but I forgot the reason why, now I remember thanks :)